src/NuGet.Core/NuGet.Build.Tasks/StaticGraphRestoreTaskBase.cs

```diff
@@ -97,34 +103,75 @@ public override bool Execute()
                 {
+                    bool errorLogged = false;
+
                     process.EnableRaisingEvents = true;
                     process.StartInfo = new ProcessStartInfo
                     {
-                        Arguments = $"\"{string.Join("\" \"", GetCommandLineArguments())}\"",
+                        Arguments = GetCommandLineArguments(globalProperties),
                         CreateNoWindow = true,
                         FileName = GetProcessFileName(ProcessFileName),
+                        RedirectStandardError = true,
                         RedirectStandardInput = true,
                         RedirectStandardOutput = true,
+#if !NETFRAMEWORK
+                        StandardInputEncoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false),
+#endif
                         UseShellExecute = false,
                         WorkingDirectory = Environment.CurrentDirectory,
                     };
 
                     // Place the output in the queue which handles logging messages coming through on StdOut
                     process.OutputDataReceived += (sender, args) => loggingQueue.Enqueue(args?.Data);
 
+                    process.ErrorDataReceived += (_, args) =>
+                    {
+                        if (args.Data != null)
+                        {
+                            // Any message on the standard error stream should be logged as an error
+                            Log.LogError(args.Data);
+
+                            errorLogged = true;
+                        }
+                    };
+
                     process.Exited += (sender, args) => semaphore.Release();
 
                     try
                     {
                         Log.LogMessageFromResources(MessageImportance.Low, nameof(Strings.Log_RunningStaticGraphRestoreCommand), process.StartInfo.FileName, process.StartInfo.Arguments);
 
-                        process.Start();
+                        Encoding previousConsoleInputEncoding = Console.InputEncoding;
 
-                        process.BeginOutputReadLine();
+                        // Set the input encoding to UTF8 without a byte order mark, the spawned process will use this encoding on .NET Framework
+                        Console.InputEncoding = new UTF8Encoding(encoderShouldEmitUTF8Identifier: false);
+
+                        try
+                        {
+                            process.Start();
+
+                            process.BeginOutputReadLine();
+                            process.BeginErrorReadLine();
+
+                            if (SerializeGlobalProperties)
+                            {
+                                using var writer = new BinaryWriter(process.StandardInput.BaseStream, Encoding.UTF8, leaveOpen: true);
+
+                                WriteGlobalProperties(writer, globalProperties);
+                            }
+
+                            process.StandardInput.Close();
```

Comment:
I feel like this should be in a `finally` block; otherwise, isn't there a chance that the process will wait for more input and not exit?

Reply:
Yeah might as well.